app/logic/Mail/EWS/EWSAccount.ts

```diff
@@ -269,9 +274,14 @@ export class EWSAccount extends MailAccount {
         await this.oAuth2.reset();
         await this.oAuth2.login(false); // will throw error, if interactive login is needed
         return await this.callEWS(aRequest); // repeat the call
+      } else if (this.NTLMauthorization && !authorization) {
+        return await this.callEWS(aRequest, await appGlobal.remoteApp.createType3MessageFromType2Message(response.WWWAuthenticate, this.username, this.password));
```

Comment:
Could you avoid this new parameter? E.g. could you set `this.NTLMauthorization` instead?

Reply:
The way NTLM works is that you have to make each request twice, the first with a static challenge and the second with a computed response to the server's challenge. I save the static challenge string in `this.NTLMauthorization` and then when I get the server's challenge I can then send the computed response.

Reply:
Yes, I understood that. I meant the way you're passing the challenges and responses within your code.
Instead of the new parameter to `callEWS()`, you could use 1 or 2 object properties, like you're already using NTLMauthorization, you could use either re-use the same property, or make a second one.

Actually, what would be even nicer would be to make a class NTLM, which computes the `Authorization:` HTTP header, in the similar way the OAuth2 class works. The API to the EWS class is only the 1 header value, which the EWS class takes from the NTLM class, and the NTLM class has all the rest of the logic.

Reply:
I've tried to implement this in branch `ben/ntlm` = #343 , but it's not finished. You should see the idea, though.

Reply:
Yes, and it won't work, because for each POST request you first have to send the type 1 message, get a type 2 response, and then repeat the request with a type 3 message which will be unique to that request. Type 3 messages cannot be reused for multiple requests!

Reply:
Are you sure that we cannot reuse the `Authentication` header? From how I read the protocol, we can.

Reply:
I already replied to your other comment.